File: BE/src/main/java/clothes/entity/Color.java
package clothes.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "color")
public class Color {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(name = "color")
    private String color;
    @Column(name = "original_price")
    private float originalPrice;
    @Column(name = "selling_price")
    private float sellingPrice;
    @Column(name = "url_image")
    private String urlImage;

    @ManyToOne
    @JoinColumn(name = "id_clothes")
    private Clothes clothes;

    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public String getColor() {
        return color;
    }
    public void setColor(String color) {
        this.color = color;
    }
    public float getOriginalPrice() {
        return originalPrice;
    }
    public void setOriginalPrice(float originalPrice) {
        this.originalPrice = originalPrice;
    }
    public float getSellingPrice() {
        return sellingPrice;
    }
    public void setSellingPrice(float sellingPrice) {
        this.sellingPrice = sellingPrice;
    }
    public String getUrlImage() {
        return urlImage;
    }
    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }

    public void setClothes(Clothes clothes) {
        this.clothes = clothes;
    }
}

File: FE/src/app/entity/color.ts
import { Clothes } from "./clothes"

export class Color {
    id : number
    color : string
    originalPrice : any
    sellingPrice : any
    urlImage : string
    constructor(
        id : number = 0,
        color : string = "",
        originalPrice : number = 0,
        sellingPrice : number = 0,
        urlImage : string = "",
    ){
        this.id = id,
        this.color = color
        this.originalPrice = originalPrice
        this.sellingPrice = sellingPrice
        this.urlImage = urlImage
    }
}

